app/src/lib/stores/app-store.ts

```diff
@@ -2837,40 +2841,48 @@ export class AppStore extends TypedBaseStore<IAppState> {
     const { changesState, branchesState } = this.repositoryStateCache.get(
       repository
     )
-    const { tip } = branchesState
-    const currentBranch = tip.kind === TipState.Valid ? tip.branch : null
-    const hasChanges = changesState.workingDirectory.files.length > 0
 
-    if (
-      enableStashing() &&
-      hasChanges &&
-      currentBranch !== null &&
-      uncommittedChangesStrategy ===
-        UncommittedChangesStrategy.askForConfirmation
-    ) {
-      this._showPopup({
-        type: PopupType.StashAndSwitchBranch,
-        branchToCheckout: foundBranch,
-        repository,
-      })
-
-      return repository
-    }
+    let transientStashCreated = false
```

Comment:
I'm not a fan of this function, `_checkoutBranch` in its current form because it's doing too much: checking out a branch, asking to stash, stashing changes on current branch, popping transient stash entry. I think it's worthing thinking about how to split this logic up so that it's clearer, but I don't think this PR is the place for that. So if you're reviewing and think this logic is complex, know that I likely agree with you, but I think it's mostly fine for now since it's all grouped together behind a check for `enableStashing`

Reply:
captured in #7607